pkg/cache/remote.go

```diff
@@ -23,10 +25,28 @@ type RemoteCache struct {
 type RemoteURL string
 
 // NewRemoteCache is the factory method for RemoteCache
-func NewRemoteCache(url RemoteURL, customHeaders http.Header) cache.ArtifactCache {
+func NewRemoteCache(url RemoteURL, customHeaders http.Header) (cache.ArtifactCache, error) {
 	ctx := client.WithCustomHeaders(context.Background(), customHeaders)
-	c := rpcCache.NewCacheProtobufClient(string(url), &http.Client{})
-	return &RemoteCache{ctx: ctx, client: c}
+
+	var err error
+	httpsInsecure := false
+	value, exists := os.LookupEnv("TRIVY_INSECURE")
```

Comment:
I think looking up env in many places makes maintenance hard. We might want to add `-insecure` here.
https://github.com/aquasecurity/trivy/blob/c275a841fd0ef0429684185a76bc85a5e72fa843/pkg/commands/app.go

Reply:
Thanks for the pointer.
Will have a look at how to rewrite this.

Reply:
I hope, I found all the needed pieces.

Please have a look.